src/commands/force/org/beta/create.ts

```diff
@@ -91,19 +91,61 @@ export class Create extends SfdxCommand {
     this.logger.debug('Create started with args %s ', this.flags);
 
     if (this.flags.type === OrgTypes.Sandbox) {
-      if (this.flags.retry !== 0) {
-        throw SfdxError.create('@salesforce/plugin-org', 'create', 'retryIsNotValidForSandboxes');
-      }
-      if (this.flags.clientid) {
-        this.ux.warn(messages.getMessage('clientIdNotSupported', [this.flags.clientid]));
-      }
+      this.validateSandboxFlags();
       return this.createSandbox();
     } else {
       // default to scratch org
       this.createScratchOrg();
     }
   }
 
+  private validateSandboxFlags(): void {
+    if (!this.flags.targetusername) {
+      throw SfdxError.create('@salesforce/plugin-org', 'create', 'requiresUsername');
+    }
+    if (this.flags.retry !== 0) {
+      throw SfdxError.create('@salesforce/plugin-org', 'create', 'retryIsNotValidForSandboxes');
+    }
+
+    if (this.flags.clientid) {
+      this.ux.warn(messages.getMessage('clientIdNotSupported', [this.flags.clientid]));
+    }
+    if (this.flags.nonamespace) {
+      this.ux.warn(messages.getMessage('noNamespaceNotSupported', [this.flags.nonamespace]));
+    }
+    if (this.flags.noancestors) {
+      this.ux.warn(messages.getMessage('noAncestorsNotSupported', [this.flags.noancestors]));
+    }
+    if (this.flags.durationdays) {
+      this.ux.warn(messages.getMessage('durationDaysNotSupported', [this.flags.durationdays]));
+    }
+  }
+
+  private createSandboxRequest(): SandboxRequest {
+    const sandboxDefFileContents = this.readJsonDefFile();
+    this.logger.debug('Create Varargs: %s ', this.varargs);
+    // the API has keys defined in capital camel case, while the definition schema has them as lower camel case
+    // we need to convert lower camel case to upper before merging options so they will override properly
+    Object.keys(sandboxDefFileContents).map((key) => {
+      const capitalKey = key.charAt(0).toUpperCase() + key.slice(1);
+      sandboxDefFileContents[capitalKey] = sandboxDefFileContents[key];
+      delete sandboxDefFileContents[key];
+    });
+    // varargs override file input
+    const sandboxReq: SandboxRequest = { SandboxName: undefined, ...sandboxDefFileContents, ...this.varargs };
+
+    if (!sandboxReq.SandboxName) {
+      // sandbox names are 10 chars or less, a radix of 36 = [a-z][0-9]
+      // technically without querying the production org, the generated name could already exist, but the chances of that are lower than the perf penalty of querying and verifying
+      sandboxReq.SandboxName = `SBX${Date.now().toString(36).slice(-7)}`;
```

Comment:
can these the `sbx` be lowercased?

Because the sandbox name is used to derive the username, which needs to be all lowercase.  
https://github.com/forcedotcom/cli/issues/1027

Reply:
nice solution for unique naming, though!